Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -210,9 +211,10 @@ def self.check_http_content(url, user_agents: [:default])
     "The URL #{url} could use HTTPS rather than HTTP"
   end
 
-  def self.http_content_headers_and_checksum(url, user_agent: :default)
+  def self.http_content_headers_and_checksum(url, hash_needed = false, user_agent: :default)
+    header_strategy = hash_needed ? "--include" : "-I"
```

Comment:
@broder Do you think it would be acceptable to simply use `--include --max-time 25` in the place of `--head` to provide more reliability? This would still eliminate the possibility of `audit --online` hanging on HTTPS urls, yet also address your concerns.